ironic-common.sh

```diff
@@ -11,6 +11,12 @@ function wait_for_interface_or_ip() {
       export IRONIC_IP=$(ip -br addr show | grep "${PROVISIONING_IP}" | grep -Po "[^\s]+/[0-9]+" | sed -e 's%/.*%%' | head -n 1)
       sleep 1
     done
+    # When an interface has multiple IP addresses, having IRONIC_IP set at this point means that the desired provisioning ip is set on the
+    # interface. However, the address returned might not be the desired one (no control over the order), so setting it back to the
+    # desired IP
+    if [ ! -z "${IRONIC_IP}" ]; then
+      export IRONIC_IP=${PROVISIONING_IP}
```

Comment:
I hit this error when trying out the patch with metal3
```
[root@master-1 core]# crictl logs 9e349ef4d8fdb |& grep 172.22.0.3                                                                                                                                                
Waiting for 172.22.0.3/24 to be configured on an interface
2020-09-23 14:41:03.656 1 DEBUG oslo_service.service [-] my_ip                          = 172.22.0.3/24 log_opt_values /usr/lib/python3.6/site-packages/oslo_config/cfg.py:2585                                   
2020-09-23 14:41:03.677 1 DEBUG oslo_service.service [-] conductor.bootloader           = http://172.22.0.3/24:6180/uefi_esp.img log_opt_values /usr/lib/python3.6/site-packages/oslo_config/cfg.py:2593          
2020-09-23 14:41:03.681 1 CRITICAL ironic [-] Unhandled error: ValueError: 172.22.0.3/24 is not IPv4 or IPv6 address                                                                                              
2020-09-23 14:41:03.681 1 ERROR ironic ValueError: 172.22.0.3/24 is not IPv4 or IPv6 address
```
I guess its down to PROVISIONING_IP being a CIDR
    - name: PROVISIONING_IP
      value: 172.22.0.3/24

Reply:
That's weird, considering https://github.com/metal3-io/baremetal-operator/blob/master/tools/run_local_ironic.sh#L26 where that variable comes from https://github.com/metal3-io/metal3-dev-env/blob/master/lib/network.sh#L56 . PROVISIONING_IP is set to 172.22.0.2 . It does not have a CIDR while being passed down to the container. Are you using metal3-dev-env or something specific for Openshift ?

Reply:
My guess is that the PROVISIONING_IP environment parameter is used differently in the Openshift version of Metal3-dev-env ? it sounds weird that something called PROVISIONING_IP actually contains a CIDR

Reply:
Yup, looks like in openshift its coming from the machine-api-operator, here
https://github.com/openshift/machine-api-operator/blob/1a2c56d/pkg/operator/baremetal_pod.go#L535
which gets a CIDR
https://github.com/openshift/machine-api-operator/blob/1a2c56d/pkg/operator/baremetal_config.go#L178
the same thing is passed into multiple different containers, I'll see what happens if I change it

Reply:
I have added a sed command to remove the /24 in case it is present to maintain the compatibility for Openshift use case. please take a look

Reply:
thanks, I'll fix this in the other components but maintaining the sed for the moment helps